railties/lib/rails/generators/app_base.rb

```diff
@@ -633,7 +633,11 @@ def target_rails_prerelease(self_command = "new")
 
           run_bundle
 
-          @argv[0] = destination_root
+          if @argv[0].start_with?("-") # rails new --some --flags APP_PATH
+            @argv.insert(0, destination_root)
+          else # rails new APP_PATH --some --flags --or --no --flags
+            @argv[0] = destination_root
+          end
```

Comment:
The purpose of `@argv[0] = destination_root` is to replace a potentially relative path with an absolute path, since we `cd` to that path before calling `target_rails_prerelease`:

https://github.com/rails/rails/blob/3dd5d4900d01c0b25f44c2fbf6ce261d367a490a/railties/lib/rails/generators/rails/app/app_generator.rb#L325-L327

https://github.com/rails/rails/blob/3dd5d4900d01c0b25f44c2fbf6ce261d367a490a/railties/lib/rails/generators/app_base.rb#L246-L247

https://github.com/rails/rails/blob/3dd5d4900d01c0b25f44c2fbf6ce261d367a490a/railties/lib/rails/generators/app_base.rb#L233-L237

Therefore, it should suffice to remove the original path (wherever it is in `@argv`) and prepend the absolute path:

```suggestion
          @argv.delete_at(@argv.index(app_path))
          @argv.unshift(destination_root)
```

Reply:
Yep that's a great idea.